### PR TITLE
[toolformer] promote svs from a topic for rag

### PR DESCRIPTION
**Pull request by chejennifer**
Promote svs that are in a topic but show up immediately after that topic to before the topic

### CODE REVIEW

File: server/lib/nl/fulfillment/fulfiller.py [hunk 1]
```diff
@@ -134,6 +135,9 @@ def fulfill(uttr: Utterance) -> PopulateState:
     # only do toolformer rig updates in single sv case
     if state.uttr.mode == QueryMode.TOOLFORMER_RIG:
       _update_chart_vars_for_rig(state)
+    # only do toolformer rig updates in single sv case
```
Comment (pradh) [inline server/lib/nl/fulfillment/fulfiller.py:138]: nit: duplicate comment?  also drop "rig" since

File: server/lib/nl/fulfillment/fulfiller.py [hunk 2]
```diff
@@ -294,3 +298,49 @@ def _update_chart_vars_for_rig(state: PopulateState):
...
+    for cv in cv_list:
+      topic_svs.update(cv.svs)
+
+    # add immediate next svs that are part of the current topic so that they are
+    # promoted to before the current topic.
+    for next_var_idx in range(idx + 1, len(ordered_cv_map)):
+      next_var, next_var_cv_list = ordered_cv_map[next_var_idx]
+      # break if not a sv in the topic because we only want to add the immediate
+      # next svs
+      if next_var.startswith(_TOPIC_PREFIX) or not next_var in topic_svs:
```
Comment (pradh) [inline server/lib/nl/fulfillment/fulfiller.py:334]: is the first check redundant?

**Review by pradh — APPROVED** [review]
Nice!

File: server/lib/nl/fulfillment/fulfiller.py (hunk 2 above)
Comment (chejennifer) [reply server/lib/nl/fulfillment/fulfiller.py:333]: I wasn't sure if chartvar.svs can include topic vars .. if not, I can remove the first check

File: server/lib/nl/fulfillment/fulfiller.py (hunk 1 above)
Comment (chejennifer) [reply server/lib/nl/fulfillment/fulfiller.py]: removed and updated the comment above to say toolformer rig and rag

File: server/lib/nl/fulfillment/fulfiller.py [hunk 3]
```diff
@@ -256,7 +260,7 @@ def _update_chart_vars_for_rig(state: PopulateState):
     for cv in chart_vars_list:
       topic_svs.update(cv.svs)
 
-  updated_chart_vars_map: Dict[str, List[ChartVars]] = {}
+  updated_chart_vars_map = OrderedDict()
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply server/lib/nl/fulfillment/fulfiller.py]: added type info back

oh good to know, but will keep to make it clear

File: server/lib/nl/fulfillment/fulfiller.py (hunk 2 above)
Comment (pradh) [reply server/lib/nl/fulfillment/fulfiller.py:333]: I think chart_vars `svs` are variables (ensured in lib/nl/explore/topic.py), but fine to keep.

**Review by pradh — APPROVED** [review]

**Pull request merged**